pylint_odoo/test_repo/broken_module/models/broken_model.py

```diff
@@ -78,3 +78,33 @@ def my_method9(self):
     def my_method10(self):
         # A example of built-in raise without parameters
         raise ZeroDivisionError
+
+    def sql_method(self, ids, cr):
+        # This is the better way and should not be detected
+        self._cr.execute(
+            'SELECT name FROM account WHERE id IN %s', (tuple(ids),))
+        self.env.cr.execute(
+            'SELECT name FROM account WHERE id IN %s', (tuple(ids),))
+        cr.execute(
+            'SELECT name FROM account WHERE id IN %s', (tuple(ids),))
+
+    def sql_injection_method(self, cr, ids):
+        # SQL injection, bad way
+        self._cr.execute(
+            'SELECT name FROM account WHERE id IN %s' % (tuple(ids),))
+        self.env.cr.execute(
+            'SELECT name FROM account WHERE id IN %s' % (tuple(ids),))
+        cr.execute(
+            'SELECT name FROM account WHERE id IN %s' % (tuple(ids),))
+
+    def sql_injection_method3(self, ids, cr2):
+        # This cr.execute2 or cr2.execute should not be detected
+        self._cr.execute2(
+            'SELECT name FROM account WHERE id IN %s' % (tuple(ids),))
+        cr2.execute(
+            'SELECT name FROM account WHERE id IN %s' % (tuple(ids),))
+
+    def sql_injection_method5(self, ids):
+        var = 'SELECT name FROM account WHERE id IN %s'
+        values = ([1, 2, 3, ], )
+        self._cr.execute(var % values)  # sql injection too
```

Comment:
We have a coverage of 100% here

<img width="716" alt="screen shot 2016-05-18 at 8 55 01 am" src="https://cloud.githubusercontent.com/assets/6644187/15361218/3ecd15f4-1cd6-11e6-88fa-e01204435904.png">

Maybe you could be interest in review the following material:

[![asciicast](https://asciinema.org/a/26890.png)](https://asciinema.org/a/26890)

[![asciicast](https://asciinema.org/a/26889.png)](https://asciinema.org/a/26889)